streams/src/main/java/org/apache/kafka/streams/state/internals/HasNextCondition.java

```diff
@@ -16,10 +16,8 @@
  */
 package org.apache.kafka.streams.state.internals;
 
-import org.apache.kafka.streams.KeyValue;
-import org.apache.kafka.streams.state.WindowStoreIterator;
+import org.apache.kafka.streams.state.KeyValueIterator;
 
-public interface PeekingWindowIterator<V> extends WindowStoreIterator<V> {
-
-    KeyValue<Long, V> peekNext();
+interface HasNextCondition<K> {
```

Comment:
Okay, could we then rename it to sth like `SegmentedBytesStoreIteratorHasNextCondition`, or just move this interface into `SegmentedBytesStore.KeySchema`?